### PR TITLE
Ys/sc 4995

### PR DESCRIPTION
**Pull request by yuliashikhareva**
# Description

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: page-objects/createCourse.js [hunk 1]
```diff
@@ -37,9 +38,54 @@ module.exports = {
 		let nameSelector = await driver.$('[data-testid="coursename"]');
 		await nameSelector.setValue(coursename)
 	},
-	setColour: async function() {
-		let inputColor = await driver.$(courseData.elem.colorCourse);
-		await inputColor.click();
+	setColour: async function(name) {
```
Comment (Ojself) [inline page-objects/createCourse.js:41]: There's a lot of repeating code here. A cleaner approach could be:
```
const colors = ["grey","metallicGold","blue","green","darkGrey","goldenPoppy","martini","violetRed","corn"]
const child = colors.indexOf(name) + 1
if (child <= 0) {
console.warn(`you did not insert a valid color. Must be ${colors},\n you inserted ${name}`)
const color = await driver.$(`.color-picker__item:nth-child(${child})`);
await color.click();
```

File: page-objects/createCourse.js [hunk 2]
```diff
@@ -80,5 +126,191 @@ module.exports = {
...
+	clickCreateCourseBtn: async function()  {
+		let btn = await driver.$(courseData.elem.createCourseBtn);
+		await btn.click();
+		await driver.pause(1000);
+	},
+	canProceedToStage: async function(stageNum) {
+		switch (stageNum) {
+			case 2: let selector = await driver.$(courseData.elem.stage1Selector);
+			let children = await selector.$$(".//*");
+			if (children.length>0) {
```
Comment (Ojself) [inline page-objects/createCourse.js:146]: There's no need for an if-statement here.
You could:
```
return children.length>0
```
Which will be either `true` or `false`

File: page-objects/createCourse.js [hunk 3]
```diff
@@ -80,5 +126,191 @@ module.exports = {
...
+				return true;
+			} else {
+			return false};
+			case 2: let selector2 = await driver.$(courseData.elem.stage2Selector);
+			let children2 = await selector2.$$(".//*");
+			if (children2.length>0) {
```
Comment (Ojself) [inline page-objects/createCourse.js:152]: Same here.
```retrun children2.length>0```

File: page-objects/createCourse.js (hunk 1 above)
Comment (yuliashikhareva) [reply page-objects/createCourse.js:41]: Thank you, Tor, a nice point!

**Pull request merged**